File: Domain/Post.cs
namespace Domain
{
	public class Post : BaseEntity
	{
		public string Body { get; set; }
		public AppUser User { get; set; }
        public Image Image { get; set; }

		public ICollection<Like> Likes { get; set; } = new List<Like>();
		public ICollection<Comment> Comments { get; set; } = new List<Comment>();
	}
}

File: client/src/app/models/post.ts
import { Image } from "./image"
import { LikesInfo } from "./likesInfo"
import { UserDetails } from "./user"

export interface Post {
  id: string
  body: string
  createdAt: Date
  user: UserDetails | null
  image: Image | null
  likesInfo: LikesInfo | null
}